src/coreclr/jit/codegenriscv64.cpp

```diff
@@ -4042,6 +4020,23 @@ int CodeGenInterface::genCallerSPtoInitialSPdelta() const
     return callerSPtoSPdelta;
 }
 
+// Produce generic and unoptimized code for loading constant to register and dereferencing it
+// at the end
+static void emitLoadConstAtAddr(emitter* emit, regNumber dstRegister, ssize_t imm)
+{
+    ssize_t high = (imm >> 32) & 0xffffffff;
+    emit->emitIns_R_I(INS_lui, EA_PTRSIZE, dstRegister, (((high + 0x800) >> 12) & 0xfffff));
+    emit->emitIns_R_R_I(INS_addi, EA_PTRSIZE, dstRegister, dstRegister, (high & 0xfff));
+
+    ssize_t low = imm & 0xffffffff;
+    emit->emitIns_R_R_I(INS_slli, EA_PTRSIZE, dstRegister, dstRegister, 11);
+    emit->emitIns_R_R_I(INS_addi, EA_PTRSIZE, dstRegister, dstRegister, ((low >> 21) & 0x7ff));
+
+    emit->emitIns_R_R_I(INS_slli, EA_PTRSIZE, dstRegister, dstRegister, 11);
+    emit->emitIns_R_R_I(INS_addi, EA_PTRSIZE, dstRegister, dstRegister, ((low >> 10) & 0x7ff));
+    emit->emitIns_R_R_I(INS_ld, EA_PTRSIZE, dstRegister, dstRegister, (low & 0x3ff));
+}
+
```

Comment:
As I know, the reason why this unoptimized code is used (instead of `emitLoadImmediate`) is there are some fixed number for code size. Am I right? If so, I hope you keep this in an optimization to-do list for later. Thank you!

Reply:
Exactly, that's the reason. I already left _TODO_ comment on caller side: [link](https://github.com/dotnet/runtime/pull/94666/commits/a0c425645c7bf6ffdc449788351df19ed9d4ba55#diff-00a478e27ec7f1116fa2dc1a5976b1a4f6762565dc92af9526d1af11958e9a94R7202). Adapting _emitLoadImmediate_ for our purpose shouldn't be a problem, but yeah, I considered it as codegen optimization, and left it for later.

Reply:
Please update TODO format based on https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/clr-jit-coding-conventions.md#7.1.5. I think `TODO:` should be `TODO-RISCV64-CQ`.